BTable multisort (#1842)
bootstrap-vue-next/bootstrap-vue-next: apps/docs/src/docs/components/table.md

descending (largest first). Clicking on it a third time will stop sorting on the column. For
single column sorting (e.g. `multisort===false`) clicking on a differnt sortable column header will
sort that column in ascending order and clear the sort order for the previously sorted column.

You can control which column is pre-sorted and the order of sorting (ascending or descending). To
pre-specify the column to be sorted use the `sortBy` model. For single column sorting (e.g. `multisort===false`)
`sortBy` should be an array containing a single `BTableSortBy` object.

```ts
 BTableSortBy = {
  order: BTableSortByOrder // Sort order can by 'asc', 'desc' or undefined
  key: string // The field name to sort by
  comparer?: (a: string, b: string) => number
}
```

- **Ascending**: Items are sorted lowest to highest (i.e. `A` to `Z`) and will be displayed with the
  lowest value in the first row with progressively higher values in the following rows.
- **Descending**: Items are sorted highest to lowest (i.e. `Z` to `A`) and will be displayed with
  the highest value in the first row with progressively lower values in the following rows.

> 

Author: ```suggestion

By default the comparer function does a `numeric localeCompare`. If one wishes to change this, use a custom comparer function with that `BTableSortBy` element. 

To prevent the table from wiping out the comparer function, internally it will set the `order` key to `undefined`, instead of just removing the element from the `sortBy` array. i.e. `:sort-by="[]"` & `:sort-by="[key: 'someKey', order: undefined]"` behave identically. Naturally if this value is given to a server, orders of undefined should be handled.
```

sort that column in ascending order and clear the sort order for the previously sorted column.

You can control which column is pre-sorted and the order of sorting (ascending or descending). To
pre-specify the column to be sorted use the `sortBy` model. For single column sorting (e.g. `multisort===false`)
`sortBy` should be an array containing a single `BTableSortBy` object.

```ts
 BTableSortBy = {
  order: BTableSortByOrder // Sort order can by 'asc', 'desc' or undefined
  key: string // The field name to sort by
  comparer?: (a: string, b: string) => number
}
```

- **Ascending**: Items are sorted lowest to highest (i.e. `A` to `Z`) and will be displayed with the
  lowest value in the first row with progressively higher values in the following rows.
- **Descending**: Items are sorted highest to lowest (i.e. `Z` to `A`) and will be displayed with
  the highest value in the first row with progressively lower values in the following rows.

By default the comparer function does a `numeric localeCompare`. If one wishes to change this, use a custom comparer function with that `BTableSortBy` element. 

> To prevent the table from wiping out the comparer function, internally it will set the `order` key to `undefined`, instead of just removing the element from the `sortBy` array. i.e. `:sort-by="[]"` & `:sort-by="[key: 'someKey', order: undefined]"` behave identically. Naturally if this value is given to a server, orders of undefined should be handled.

Author: ```suggestion
To prevent the table from wiping out the comparer function, internally it will set the `order` key to `undefined`, instead of just removing the element from the `sortBy` array. i.e. `:sort-by="[]"` & `:sort-by="[key: 'someKey', order: undefined]"` behave identically. Naturally if this value is given to a server, orders of undefined should be handled.

```

interface SortPerson {
  first_name: string
  last_name: string
  age: number
  isActive: boolean
}

const sortItems: TableItem<Person>[] = [
  {isActive: true, age: 40, first_name: 'Dickerson', last_name: 'Macdonald'},
  {isActive: true, age: 45, first_name: 'Zelda', last_name: 'Macdonald'},
  {isActive: false, age: 21, first_name: 'Larsen', last_name: 'Shaw'},
  {isActive: false, age: 89, first_name: 'Geneva', last_name: 'Wilson'},
  {isActive: false, age: 89, first_name: 'Gary', last_name: 'Wilson'},
  {isActive: true, age: 38, first_name: 'Jami', last_name: 'Carney'},
]

const sortFields: Exclude<TableFieldRaw<SortPerson>, string>[] = [
  {key: 'last_name', sortable: true},
  {key: 'first_name', sortable: true},
  {key: 'age', sortable: true},
  {key: 'isActive', sortable: false},
]
</script>
```

  </template>
</HighlightCard>

`sorbBy` is a [named model](https://vuejs.org/guide/components/v-model.html#multiple-v-model-bindings)
so it can be bound to an object that will be updated with the current sort state when the user changes
sorting by clicking the headers.

<HighlightCard>
  <BTable
    :sort-by="singleSortBy"
    :items="sortItems"
    :fields="sortFields"/>

>     <div>singleSortBy = {{ JSON.stringify(singleSortBy) }}</div>

Author: @dwgray You don't need to JSON.stringify in templates. Vue automatically uses JSON.stringify on objects

}

const sortItems: TableItem<Person>[] = [
  {isActive: true, age: 40, first_name: 'Dickerson', last_name: 'Macdonald'},
  {isActive: true, age: 45, first_name: 'Zelda', last_name: 'Macdonald'},
  {isActive: false, age: 21, first_name: 'Larsen', last_name: 'Shaw'},
  {isActive: false, age: 89, first_name: 'Geneva', last_name: 'Wilson'},
  {isActive: false, age: 89, first_name: 'Gary', last_name: 'Wilson'},
  {isActive: true, age: 38, first_name: 'Jami', last_name: 'Carney'},
]

const sortFields: Exclude<TableFieldRaw<SortPerson>, string>[] = [
  {key: 'last_name', sortable: true},
  {key: 'first_name', sortable: true},
  {key: 'age', sortable: true},
  {key: 'isActive', sortable: false},
]
</script>
```

  </template>
</HighlightCard>

Tables can be sorted by multiple columns. Programmaticly, this can be done by adding more entries
to the `sortBy` array. From the user inteface, multi-sort works as follows:

- Clicking on a sortable header that isn't currently sorted adds it as `ascending` to the end of the sortBy list
- Clicking on a sortable header that is currently sorted as ascending makes it **descending**, but leaves it in the same order in the `sortBy` list

> - Clicking on a sortable header that is currently sorted as descending removes it from the list

Author: ```suggestion
- Clicking on a sortable header that is currently sorted as descending will set the order to undefined. If `must-sort` is `true` OR if `mustSort` is an array that contains that columns `key`, it will skip to be `ascending`
```
